### PR TITLE
Support create Asset with stream

### PR DESCRIPTION
**Pull request by Steven-Chan**
connect #258

### CODE REVIEW

File: skygear/src/main/java/io/skygear/skygear/Asset.java [hunk 1]
```diff
@@ -155,4 +169,111 @@ public JSONObject toJson() {
...
+            if (this.uri != null) {
+                return this.buildFromUri();
+            }
+
+            return null;
+        }
+
+        private Asset buildFromInputStream() {
+            ensureStateIsNonNull(this.mimeType, "MimeType");
+            ensureStateIsNonNull(this.size, "Size");
```
Comment (ben181231) [inline skygear/src/main/java/io/skygear/skygear/Asset.java]: I think `this.size` is not being set. A `public Builder setSize(long size)` method is needed in the builder.

File: skygear/src/main/java/io/skygear/skygear/Asset.java [hunk 2]
```diff
@@ -35,28 +45,43 @@
...
      * Instantiates a new Asset with name, MIME type and file data.
      *
+     * @deprecated use {@link Asset.Builder} instead.
+     *
      * @param name     the name
      * @param mimeType the MIME type
      * @param data     the data
      */
+    @Deprecated
     public Asset(String name, String mimeType, byte[] data) {
```
Comment (ben181231) [inline skygear/src/main/java/io/skygear/skygear/Asset.java]: Better update the usage of this method in test cases to use builder

File: skygear/src/main/java/io/skygear/skygear/Asset.java [hunk 3]
```diff
@@ -155,4 +169,111 @@ public JSONObject toJson() {
...
+            this.data = data;
+            return this;
+        }
+
+        public Builder setInputStream(InputStream inputStream) {
+            this.inputStream = inputStream;
+            return this;
+        }
+
+        public Asset build() {
```
Comment (ben181231) [inline skygear/src/main/java/io/skygear/skygear/Asset.java]: Better add the unit tests for asset builder for the 3 building flows.

**Comment by Steven-Chan** [comment]
@ben181231 updated

**Pull request merged**